File: Medium/Python/2149. Rearrange Array Elements by Sign.py
class Solution(object):
    def rearrangeArray(self, nums):
        # Create a new list, an even index, and an odd index variable
        rearranged = [0] * len(nums)
        i = 0 # Even index
        j = 1 # Odd index

        # Go over the list, adding positives, then negatives, to the list
        for k in nums:
            if 0 < k:
                rearranged[i] = k
                i = i + 2
            else:
                rearranged[j] = k
                j = j + 2

        # Return the new list
        return rearranged

File: Medium/Java/2149. Rearrange Array Elements by Sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        // Create a new array, an even index, and odd index variable
        int[] rearranged = new int[nums.length];
        int i = 0; // Even index
        int j = 1; // Odd index

        // Go over the array adding positives, then negatives, to the array
        for(int k = 0; k < nums.length; k++){
            if(0 < nums[k]){
                rearranged[i] = nums[k];
                i += 2;
            }else{
                rearranged[j] = nums[k];
                j += 2;
            }
        }

        // Return the new array
        return rearranged;
    }
}
